doc/usecases/kdb/UC_modify_config.md

```diff
@@ -0,0 +1,28 @@
+# Use Case: Modifying application configuration
+
+## Summary
+
+- **Title:** Modifying application configuration
+- **Scope:** `libelektra-kdb`
+- **Level:** Developer Goal
+- **Actors:** Application, Elektra
+- **Brief:** Application modifies configuration
+
+## Scenarios
+
+- **Precondition:** -
+- **Main success scenario:**
+  - [Application loads configuration](UC_load_config.md)
+  - Application modifies configuration in memory
+  - [Application stores configuration](UC_store_config.md)
```

Comment:
You can be more low-level, e.g. I don't see any problem to say which functions get called:

```suggestion
  - `kdbSet()` is called immediately afterwards.
```

Reply:
Again IMO a use case is inherently abstract and high-level. Mentioning a specific function is definitely out of scope. Everything would be the same, if we renamed `kdbSet` to `foofoo`, so why mention it? The information that `kdbSet` implements this use case should be in the documentation for `kdbSet`.

Reply:
Also applies above, but we also have to keep in mind that most use case (and these ones especially) should apply independent of language binding. Therefore, we should avoid referencing specific APIs.

Reply:
I agree the use cases are also the basis for the bindings. But I don't think that the name we use in the C API will be disturbing. We recommend for APIs to use the same names, maybe except of prefix or similar, e.g., `Elektra::Kdb.set()` but it is still clear that it is elektraKdbSet.

Reply:
Hm okay, let me phrase it this way. The proper order (that you yourself suggested) would be: (1) use case, (2) decision, (3) implementation.

I don't think at the use case stage API names should be defined. That should be part of the decision, or maybe even an implementation detail (for internal APIs). So if we imagine this use case was written before the API existed, then it the name `kdbSet` wouldn't even exist yet.

While we could make an exception here, because the name does exist and we can use it. I think that would be a bad idea. These use cases are the core part that everybody contributing to Elektra needs to understand. So if they want to write new use cases, they will look at these (because they'll understand the contents already and can focus on the format). Using API names here, may suggest they should be used in other use cases too.

So far that's not a big problem. But IMO the issue is then twofold:
1. By using names, the use case stage will be unnecessarily prolonged, because there will be an inevitable discussion about API names. This discussion contributes nothing to the development of the use case and just steals peoples focus.
2. If for whatever reason, we decide to change the name later (during the decision or implementation stage), then we would have to update the use case. For no reason other than to fix the name. I don't think this would ever be worth it, unless there is a very niche use case that is impossible to understand without concrete API names.

Reply:
It is not only about "defining names" (but for new APIs of course also that) but my main reason for the wish now is that it is simply practical. For the future: Ideal would be a link from the use case to the implementation (traceability links) and to the API documentation (for convenience). This obviously can only be done where the implementation already exist.

For now https://github.com/ElektraInitiative/libelektra/pull/4697#issuecomment-1332014001 was enough, thank you for writing it down.

> By using names, the use case stage will be unnecessarily prolonged, because there will be an inevitable discussion about API names. This discussion contributes nothing to the development of the use case and just steals peoples focus.

I strongly disagree. The name is so important and so strongly connected to its use case, that it is actually a good idea to develop it together.

> If for whatever reason, we decide to change the name later

This will probably never be done again after your refactoring. And your refactoring tool hopefully will be able to also change the names in the docu and use cases, won't it?

Reply:
> For the future: Ideal would be a link from the use case to the implementation (traceability links) and to the API documentation (for convenience).

IMO it would be better to have a file in `doc/contrib` that describes how the various use cases are covered by the API. Use cases might not map 1:1 onto the API. One use case might be multiple functions or one function may cover multiple use cases. It's easier if there is a separate file that describes the mapping.

> The name is so important and so strongly connected to its use case, that it is actually a good idea to develop it together.

Yes, the name is important. That's why IMO the API (including the names) should be developed during the decision stage, where it can be discussed and reviewed. However, I don't think it is useful to discuss the names with the use cases.

At the use case stage, the name would just be a placeholder anyway. It shouldn't matter, whether the function will be called `foo` or `bar`. What matters for the use case is what the function does.

> > If for whatever reason, we decide to change the name later
> 
> This will probably never be done again after your refactoring.

Skilfully ignored the examples in the parentheses, I see...

I didn't mean change the name of a function that is already merged and in use. That hopefully won't happen and would be a big breaking change anyway. I meant changing the name after the use case is merged, but before the implementation is written.

> your refactoring tool hopefully will be able to also change the names in the docu and use cases, won't it?

Highly unlikely. Updating code snippets in markdown files should be possible. They should be tagged as <code>\`\`\`c</code>, so I could look for that and try to parse the syntax tree. But for function names mentioned randomly, maybe not even in backticks (<code>\`</code>), I couldn't do any better than a standard regex replace.